pkg/prometheus/rules.go

```diff
@@ -122,13 +123,21 @@ func (c *Operator) createOrUpdateRuleConfigMaps(ctx context.Context, p *monitori
 		"namespace", p.Namespace,
 		"prometheus", p.Name,
 	)
-	for _, cm := range newConfigMaps {
-		_, err = cClient.Create(ctx, &cm, metav1.CreateOptions{})
+	for _, cm := range createOrUpdateConfigMaps {
+		err = k8sutil.CreateOrUpdateConfigMap(ctx, cClient, &cm)
 		if err != nil {
-			return nil, errors.Wrapf(err, "failed to create new ConfigMap '%v'", cm.Name)
+			return nil, errors.Wrapf(err, "failed to createOrUpdate ConfigMap '%v'", cm.Name)
 		}
 	}
 
+	// supple configmap volumeMount number, at least `default` number of rule configmaps mount.
```

Comment:
```suggestion
	// supple configmap volumeMount number, at least `default` number of rule configmaps mount.
	// this is to prevent the StatefulSet Pods from being frequently redeployed when the size of the
	// configuration can easily fluctuate between the maximum size of a single ConfigMaps or not.
```